cmd/ctl/pkg/factory/factory.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factory
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubectl/pkg/cmd/util"
+
+	// Load all auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+var (
+	kubeConfigFlags = genericclioptions.NewConfigFlags(true)
+	factory         = util.NewFactory(kubeConfigFlags)
+)
+
+// Factory provides a set of clients and configurations to authenticate and
+// access a target Kubernetes cluster. Factory will ensure that its fields are
+// populated and valid during command execution.
+type Factory struct {
```

Comment:
suggestion: "factory" is a fuzzy word and feels confusing to me 😞 Maybe "CmdConfig" or something like that?

Reply:
Again, this is the same naming scheme that [Kubernetes uses](https://github.com/kubernetes/kubernetes/blob/b40d9e8dd5de13a11eb5c56c8fdb7794047764d9/staging/src/k8s.io/kubectl/pkg/cmd/util/factory.go#L41).

Reply:
Would renaming the package from `factory` to `util` help?